### PR TITLE
Sprintf support in the translator

### PR DESCRIPTION
**Pull request by jtabet**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Add support for sprintf style args in translator |
| Type? | improvement |
| Category? | CO |
| BC breaks? | No |
| Deprecations? | No |
| Fixed ticket? | No |
| How to test? | Translate strings including sprintf syntax with the translator (see screenshot below) |

### CODE REVIEW

**Comment by xBorderie** [comment]
@Shine-neko Thumbs down? Why so?

**Comment by Shine-neko** [comment]
@xBorderie Another hack ...  This does not correct the problems

**Comment by julienbourdeau** [comment]
@Shine-neko This is meant to be used only with old strings in `$this->l()` and especially to keep translation from PrestaShop 1.6.

We're open to any feedbacks and better way to do this

File: src/PrestaShopBundle/Translation/SprintfTranslatorTrait.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     *
+     * @param  string  $string
+     * @return boolean
+     */
+    private final function isSprintfString($string)
+    {
+        return (bool) preg_match_all('#(?:%%|%(?:[0-9]+\$)?[+-]?(?:[ 0]|\'.)?-?[0-9]*(?:\.[0-9]+)?[bcdeufFosxX])#', $string)
```
Comment (julienbourdeau) [inline src/PrestaShopBundle/Translation/SprintfTranslatorTrait.php]: I will trust you on this 🎯

**Pull request merged**